### DISCUSSION TITLE
Maximum concurrent users?

### DISCUSSION THREAD

**Discussion by puddleglum56** [mml-io/mml#202] [General]
I'm evaluating some other frameworks for a metaverse game I am developing in Unity with NATS networking. I suspect this question has been asked before, and depends on the underlying server hardware, but is MML designed for potentially high (>1000) numbers of concurrent players in a single world? I've seen some cool demos of Improbable's tech, especially for consistent worlds with tons of players, and am wondering if MML has that capability.

**Comment by MarcusLongmuir** [comment]
Hey @puddleglum56. We haven't run massive scale tests against MML, but running an MML/NetworkedDOM document on a regular (not micro) VPS from a provider should see you easily get 50+ users connected to it. And the [3d-web-experience](https://github.com/mml-io/3d-web-experience) project which is mostly how the MML project is used to build web-based worlds should support ~50 users on a regular VPS too. I'm being conservative with these numbers, but you're unlikely to get to 1k users with the current versions of these projects. 

We've got plans to tackle high scale soon and the MML/NetworkedDOM paradigm has been designed with high scale in mind (fan out of arbitrary DOM state is logically simple). It would be useful to know what you're trying to build content complexity-wise, and the scale you're targeting.
